site/type_def.md

```diff
@@ -0,0 +1,246 @@
+                Enabled: true
+                ListenPort: 7002
+        soa_server1:
+            ListenAddress: managed1.example.com
+            ListenPort: 8001
+            Cluster: soa_cluster
+            Machine: machine2
+            SSL:
+                Enabled: true
+                ListenPort: 8002
+        soa_server2:
+            ListenAddress: managed2.example.com
+            ListenPort: 8001
+            Cluster: soa_cluster
+            Machine: machine3
+            SSL:
+                Enabled: true
+                ListenPort: 8002
+        osb_server1:
+            ListenAddress: managed1.example.com
+            ListenPort: 9001
+            Cluster: osb_cluster
+            Machine: machine2
+            SSL:
+                Enabled: true
+                ListenPort: 9002
+        osb_server2:
+            ListenAddress: managed2.example.com
+            ListenPort: 9001
+            Cluster: osb_cluster
+            Machine: machine3
+            SSL:
+                Enabled: true
+                ListenPort: 9002
+    UnixMachine:
+        machine1:
+            NodeManager:
+                ListenAddress: myadmin.example.com
+                ListenPort: 5556
+        machine2:
+            NodeManager:
+                ListenAddress: managed1.example.com
+                ListenPort: 5556
+        machine3:
+            NodeManager:
+                ListenAddress: managed2.example.com
+                ListenPort: 5556
+    SecurityConfiguration:
+        NodeManagerUsername: weblogic
+        NodeManagerPasswordEncrypted: welcome1
+```
+
+### Custom Extension Templates
+
+The `customExtensionTemplates` attribute can be used to specify custom extension templates to be applied to the domain. These should be specified as absolute file paths, and can use tokens.  
+
+```json
+{
+    "name": "MyCustom",
+    "description": "My custom type domain definitions",
+    "versions": {
+        "12.2.1.3": "My_12213"
+    },
+    "definitions": {
+        "My_12213": {
+            "baseTemplate": "Basic WebLogic Server Domain",
+            "extensionTemplates": [ ],
+            "customExtensionTemplates": [
+                "/user/me/templates/my-template.jar",
+                "@@ORACLE_HOME@@/user_templates/other-template.jar"
+            ],
+            "serverGroupsToTarget": [ "MY-MAN-SVR" ],
+            "rcuSchemas": [ ]
+        }
+    }
+}
+```
+
+If there are any server groups in the custom template that should be targeted to managed servers, they should be specified in the `serverGroupsToTarget` attribute, similar to `MY_MAN_SVR` in the example above.
+
+### Targeting in Earlier Weblogic Server Versions
+
+Templates in WebLogic Server versions prior to 12.2.1 may require use of the `applyJRF` WLST command to correctly target resources to the correct clusters and servers. The default behavior for WebLogic Deploy Tooling is to invoke `applyJRF` only when the `extensionTemplates` list includes JRF templates.
```

Comment:
may require use of ->  may require the use of

Reply:
Fixed